framework/src/meshgenerators/PatternedMeshGenerator.C

```diff
@@ -25,12 +26,9 @@ validParams<PatternedMeshGenerator>()
   InputParameters params = validParams<MeshGenerator>();
 
   params.addRequiredParam<std::vector<MeshGeneratorName>>("inputs", "The input MeshGenerators.");
-  params.addRangeCheckedParam<Real>(
-      "x_width", 0, "x_width>=0", "The tile width in the x direction");
-  params.addRangeCheckedParam<Real>(
-      "y_width", 0, "y_width>=0", "The tile width in the y direction");
-  params.addRangeCheckedParam<Real>(
-      "z_width", 0, "z_width>=0", "The tile width in the z direction");
+  params.addRangeCheckedParam<Real>("x_width", "x_width>=0", "The tile width in the x direction");
+  params.addRangeCheckedParam<Real>("y_width", "y_width>=0", "The tile width in the y direction");
+  params.addRangeCheckedParam<Real>("z_width", "z_width>=0", "The tile width in the z direction");
```

Comment:
You can provide a "checking function" (like you've done here) and a default value. Then you don't have to test for `isParamValid` later.